docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,378 @@
+//
+// The format should match standard CIDR notation (for example, "10.128.0.0/16").
+// This field must be omitted if `ipam.mode` is `Disabled`.
+//
+// +optional
+Subnets DualStackCIDRs `json:"subnets,omitempty"`
+
+// JoinSubnets are used inside the OVN network topology.
+//
+// Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+// This field is only allowed for "Primary" network.
+// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+// When omitted, the platform will choose a reasonable default which is subject to change over time.
+//
+// +optional
+JoinSubnets DualStackCIDRs `json:"joinSubnets,omitempty"`
+
+// DefaultGatewayIPs are is the default gateway IP used in the internal OVN topology.
+//
+// Dual-stack clusters may set 2 IPs (one for each IP family), otherwise only 1 IP is allowed.
+// This field is only allowed for "Primary" network.
+// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+// When omitted, the first IP address from the pod network subnet is used.
+//
+// +optional
+DefaultGatewayIPs DualStackIPs `json:"defaultGatewayIPs,omitempty"`
+
+// NodeManagementIPs are is the node management IP used in the internal OVN topology.
+//
+// Dual-stack clusters may set 2 IPs (one for each IP family), otherwise only 1 IP is allowed.
+// This field is only allowed for "Primary" network.
+// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+// When omitted, the second IP address from the pod network subnet is used.
+//
+// +optional
+NodeManagementIPs DualStackIPs `json:"nodeManagementIPs,omitempty"`
+
+// IPAM section contains IPAM-related configuration for the network.
+// +optional
+IPAM *IPAMConfig `json:"ipam,omitempty"`
+}
+
+// +kubebuilder:validation:XValidation:rule="isIP(self)", message="IP is invalid"
+type IP string
+
+// +kubebuilder:validation:MinItems=1
+// +kubebuilder:validation:MaxItems=2
+// +kubebuilder:validation:XValidation:rule="size(self) != 2 || !isIP(self[0]) || !isIP(self[1]) || ip(self[0]).family() != ip(self[1]).family()", message="When 2 IPs are set, they must be from different IP families"
+type DualStackIPs []IP
+
+```
+
+#### IPAMClaim API changes
+
+- TODO: Add status conditions to the IPAMClaim CRD, this will be an external enhancement in the network-plumbing group
+
+### Implementation Details
+
+#### Configurability
+
+The changes outlined in this enhancement should be configurable. This means a configuration knob
+is required to instruct OVN-Kubernetes on whether to process the annotation described in the 
+[Pod network identity](#pod-network-identity) section.
+
+#### NetworkSelectionElement annotation
+
+The `v1.multus-cni.io/default-network` annotation should only be processed for new pods, 
```

Comment:
this annotation I think is already used in our code today and potentially bypasses the lookup for a UDN:
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/go-controller/pkg/util/multi_network.go#L1277

obviously we can change the code, but i just wanted to make sure with this enhancement proposal we are not regressing or breaking any previous behavior used by this annotation.

Reply:
This annotation was only used for cluster default network, I've added a sentence to clarify it.